flink-streaming-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaConsumer.java

```diff
@@ -640,30 +642,38 @@ public PeriodicOffsetCommitter(long commitInterval, FlinkKafkaConsumer consumer)
 		@Override
 		public void run() {
 			try {
+
 				while (running) {
 					try {
 						Thread.sleep(commitInterval);
 						//  ------------  commit current offsets ----------------
 
 						// create copy of current offsets
-						long[] currentOffsets = Arrays.copyOf(consumer.lastOffsets, consumer.lastOffsets.length);
-
-						Map<TopicPartition, Long> offsetsToCommit = new HashMap<>();
-						for (TopicPartition tp : (List<TopicPartition>)consumer.subscribedPartitions) {
-							int partition = tp.partition();
-							long offset = currentOffsets[partition];
-							long lastCommitted = consumer.commitedOffsets[partition];
-
+						HashMap<KafkaTopicPartition, Long> currentOffsets = (HashMap<KafkaTopicPartition, Long>) consumer.lastOffsets.clone();
+
+						Map<KafkaTopicPartition, Long> offsetsToCommit = new HashMap<>();
+						for (KafkaTopicPartition tp : (List<KafkaTopicPartition>)consumer.subscribedPartitions) {
```

Comment:
A bit unclean to cast here from `List<KafkaTopicPartitionLeader>` to `List<KafkaTopicPartition>`. Why not simply iterate with the `KafkaTopicPartitionLeader` type?